### PR TITLE
EVG-7210: Render Logs column in task page tests table

### PR DESCRIPTION
**Pull request by SupaJoon**
This PR renders the logs column buttons and relies on https://github.com/evergreen-ci/evergreen/pull/3176.
Screen Shot 2020-02-19 at 5 37 37 PM

My tests don't include verifying that a new tab has been opened after clicking on a Log column button because Cypress docs recommends against it. Instead they recommend to test the existence of attributes that would cause a new tab to open (https://docs.cypress.io/guides/references/trade-offs.html#Multiple-tabs). 

The `Buttons in log column should have target=_blank attribute` test may not pass until I insert some test results with log urls in our test server.

### CODE REVIEW

File: src/pages/task/TestsTableCore.tsx [hunk 1]
```diff
@@ -77,17 +81,60 @@ const columns: Array<ColumnProps<TaskTestsData>> = [
...
+    }: {
+      htmlDisplayURL: string;
+      rawDisplayURL: string;
+    }): JSX.Element => {
+      return (
+        <>
+          {htmlDisplayURL && (
+            <ButtonWrapper>
+              <Button
+                className="htmlBtn"
```
Comment (SupaJoon) [inline src/pages/task/TestsTableCore.tsx]: This className is applied so cypress can find this button

File: src/pages/task/TestsTableCore.tsx [hunk 2]
```diff
@@ -77,17 +81,60 @@ const columns: Array<ColumnProps<TaskTestsData>> = [
...
+                variant="default"
+                href={htmlDisplayURL}
+              >
+                HTML
+              </Button>
+            </ButtonWrapper>
+          )}
+          {rawDisplayURL && (
+            <Button
+              className="rawBtn"
```
Comment (SupaJoon) [inline src/pages/task/TestsTableCore.tsx]: This className is applied so cypress can find this button

File: src/pages/task/TestsTableCore.tsx (hunk 1 above)
Comment (tgrander) [reply src/pages/task/TestsTableCore.tsx]: what about using ids for cypress. we would only use ids for testing which makes it clear it's not for styling

File: src/pages/task/TestsTableCore.tsx (hunk 2 above)
Comment (tgrander) [reply src/pages/task/TestsTableCore.tsx]: what about using ids for cypress. we would only use ids for testing which makes it clear it's not for styling

File: src/pages/task/TestsTableCore.tsx [hunk 3]
```diff
@@ -32,6 +33,9 @@ const SpinWrapper = styled.div({
   paddingBottom: 40,
   border: "1px solid #e8e8e8"
 });
+const ButtonWrapper = styled.span({
+  marginRight: 8
+});
```
Comment (tgrander) [inline src/pages/task/TestsTableCore.tsx]: styled components should always go at bottom of file for consistency

**Comment by SupaJoon** [comment]
> i've read about opening things in new tabs as being an anti pattern in UX. would like to get some discussion going around this as it relates to a few parts of the app. opening in new app breaks user flow, for example they are not able to use the 'back' button to go back to the previous page anymore. and if a user does wish to open in a new tab then they can always use `cmd + click` on the link

Ben recommends opening the Logs in a new tab but I see the benefit of giving the user both options.

File: src/pages/task/TestsTableCore.tsx (hunk 2 above)
Comment (SupaJoon) [reply src/pages/task/TestsTableCore.tsx]: Assigning an id to button will result in duplicate id's and I would like to keep all id's unique so we are within the HTML 5.1 spec. It also seems overkill to assign a classname just for testing purposes and I think I need to remove the classname and improve the selector in cypress.

**Comment by john-m-liu** [comment]
evergreen retry

File: src/pages/task/TestsTableCore.tsx (hunk 2 above)
Comment (john-m-liu) [reply src/pages/task/TestsTableCore.tsx]: Is there a test case where you'd want to find the logs button in a specific row though? If that's the case I think it makes sense to generate an ID that contains the row index

Comment (tgrander) [reply src/pages/task/TestsTableCore.tsx]: > Assigning an id to button will result in duplicate id's and I would like to keep all id's unique so we are within the HTML 5.1 spec. It also seems overkill to assign a classname just for testing purposes and I think I need to remove the classname and improve the selector in cypress.

just append an id to the end of each id

**Comment by tgrander** [comment]
The test failures on this PR look like they fail because of actual errors in the code, which is great news considering our test failures on FE were meaningless for so long. i'm sure you're working on those and i will approve once i see dat green.

**Pull request merged**